File: 백준/Silver/10828. 스택/스택.java
import java.util.*;
import java.io.*;

public class Main {
    private static void execute(String line, Stack<Integer> stack) {
        String[] command = line.split(" ");
        
        switch(command[0]) {
            case "push":
                stack.push(Integer.parseInt(command[1]));
                break;
            case "pop":
                System.out.println(stack.empty() ? -1 : stack.pop());
                break;
            case "size":
                System.out.println(stack.size());
                break;
            case "empty":
                System.out.println(stack.empty() ? 1 : 0);
                break;
            case "top":
                System.out.println(stack.empty() ? -1 : stack.peek());
                break;
        }
    }
    
    public static void main(String args[]) throws IOException {
        Stack<Integer> stack = new Stack<>();
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        int n = Integer.parseInt(br.readLine());
        for(int i = 0; i < n; i++) {
            execute(br.readLine(), stack);
        }
    }
}

File: 백준/Silver/10828. 스택/스택.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val sb = StringBuilder()
    val n = br.readLine().toInt()
    val stack = ArrayDeque<Int>()

    repeat(n) {
        val command = br.readLine().split(" ")
        when (command[0]) {
            "push" -> {
                stack.addLast(command[1].toInt())
            }
            "pop" -> {
                sb.append(stack.removeLastOrNull() ?: -1).append('\n')
            }
            "size" -> {
                sb.append(stack.size).append('\n')
            }
            "empty" -> {
                sb.append(if (stack.isEmpty()) 1 else 0).append('\n')
            }
            "top" -> {
                sb.append(stack.lastOrNull() ?: -1).append('\n')
            }
        }
    }
    print(sb)
}